### PR TITLE
fix: [CDS-78674]: expanded executions in the list page should stay expanded even if new executions are added

### PR DESCRIPTION
**Pull request by Eric761**
### Summary

Ticket - https://harness.atlassian.net/browse/CDS-78674

- Expanded executions in the list page should stay expanded even if new executions are added

##
**Root Cause -**

React-table internally uses an `expanded` array to determine which rows are expanded based on indexing. 
##
Screenshot 2023-09-07 at 3 37 45 PM

##
**Solution -**

Polling occurs on the execution list page, so we must keep a **set data structure** to store the execution ids of expanded rows, and then check if the row is expanded and call - `row.toggleRowExpanded(expanded?: boolean)` to set the expanded state of a row.

Link - [Expanding | TanStack Table Docs](https://tanstack.com/table/v8/docs/api/features/expanding#toggleexpanded)

#### Demo Video

**Before -** 

https://github.com/harness/harness-core-ui/assets/73639897/8b9e0b48-a591-4465-97fc-a89b71991e39

##
**After -**

`Re-run case :`

https://github.com/harness/harness-core-ui/assets/73639897/c397d302-7589-4fd3-bb18-d0b2546c4afe

`Retry case :`

https://github.com/harness/harness-core-ui/assets/73639897/645e4884-c497-4f0f-8d2f-83d10878518b

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/70-pipeline/pages/execution-list/ExecutionListTable/ExecutionListCells.tsx [hunk 1]
```diff
@@ -172,13 +172,43 @@ export const RowSelectCell: CellType = ({ row }) => {
   )
 }
 
-export const ToggleAccordionCell: Renderer<{ row: UseExpandedRowProps<PipelineExecutionSummary> }> = ({ row }) => {
+export const ToggleAccordionCell: Renderer<{
```
Comment (ishantmehtaharness) [inline src/modules/70-pipeline/pages/execution-list/ExecutionListTable/ExecutionListCells.tsx:175]: Do we want jest test cases?

**Review by ishantmehtaharness — APPROVED** [review]
Looks Good

File: src/modules/70-pipeline/pages/execution-list/ExecutionListTable/ExecutionListCells.tsx (hunk 1 above)
Comment (Eric761) [reply src/modules/70-pipeline/pages/execution-list/ExecutionListTable/ExecutionListCells.tsx:175]: Not at the moment @ishantmehtaharness, as this is linked to the polling of the listing API, which requires further testing. Will investigate the possibility of testing this using cypress and plan to include it in future PRs.

**Review by PratyushGar — APPROVED** [review]

File: src/modules/70-pipeline/pages/execution-list/ExecutionListTable/ExecutionListCells.tsx [hunk 2]
```diff
@@ -58,7 +58,7 @@ import { DateTimeContent } from '@common/components/TimeAgoPopover/TimeAgoPopove
 import { useNotesModal } from '@pipeline/pages/execution/ExecutionLandingPage/ExecutionHeader/NotesModal/useNotesModal'
 import FrozenExecutionDrawer from './FrozenExecutionDrawer/FrozenExecutionDrawer'
 import { CITriggerInfo, CITriggerInfoProps } from './CITriggerInfoCell'
-import type { ExecutionListColumnActions } from './ExecutionListTable'
+import type { ExecutionListColumnActions, ExecutionListExpandedColumnProps } from './ExecutionListTable'
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/pages/execution-list/ExecutionListTable/ExecutionListCells.tsx:61]: Is this is used only here then we should declare it in this file.

Comment (Eric761) [reply src/modules/70-pipeline/pages/execution-list/ExecutionListTable/ExecutionListCells.tsx:61]: sure, will include it in upcoming PR.

**Pull request merged**